astropy/table/table.py

```diff
@@ -261,7 +265,11 @@ def as_array(self, keep_byteorder=False):
 
         dtype = []
 
-        cols = self.columns.values()
+        if names != None:
+            cols = self[names].columns.values()
```

Comment:
@himanshupathak21061998 - sorry, I made this comment a few days ago but must not have actually put it it in the review...  So one more fix, related to performance.  Using `self[names]` is simple and clean, but it is inefficient since this makes a temporary full copy of all the data for `names` and creates a new table just to get out the subset of columns.  Instead:
```
cols = self.columns.values()

if names != None:
    cols = [col for col in cols if col.info.name in names]
```

Reply:
Thanks for reviewing  suggested changes have been performed